### PR TITLE
macros: improve dialog macros to use the dialog labels

### PR DESCRIPTION
**Pull request by kemenaran**
This allow to reference a dialog by label (rather than with a numeric constants). It makes easier to search where a specific dialog is used in the code base.

Macro code written by @daid.

TODO: use ld_dialog_low where relevant (see #440)

### CODE REVIEW

File: src/code/home/dialog.asm [hunk 1]
```diff
@@ -83,6 +83,7 @@ OpenDialogInTable2::
 ; Input:
 ;   a: dialog index in table 0
 OpenDialog::
```
Comment (daid) [inline src/code/home/dialog.asm:85]: Shouldn't we just remove the "old" label? It always kind annoyed me that `OpenDialog` only works for dialogs 0-FF

Comment (kemenaran) [reply src/code/home/dialog.asm:85]: Yes, me too. The thing is, it is still referenced by OpenDialogInTable1 and OpenDialogInTable2, but this reference could be a local label instead.

Comment (kemenaran) [reply src/code/home/dialog.asm]: Done, simply by getting rid of OpenDialog, and documenting what goes on for variants.

**Comment by kemenaran** [comment]
I moved the "using ld_dialog_low" part to another issue and PR, so this one is standalone now.

**Pull request merged**